packages/react/src/components/IconIndicator/index.tsx

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright IBM Corp. 2016, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import cx from 'classnames';
+import { usePrefix } from '../../internal/usePrefix';
+import {
+  ErrorFilled,
+  CheckmarkFilled,
+  WarningAltFilled,
+  WarningAltInvertedFilled,
+  UndefinedFilled,
+  InProgress,
+  Incomplete,
+  CircleDash,
+  UnknownFilled,
+  WarningSquareFilled,
+  CheckmarkOutline,
+  PendingFilled,
+} from '@carbon/icons-react';
+
+export const IconIndicatorKinds = [
+  'error',
+  'caution-major',
+  'caution-minor',
+  'undefined',
+  'succeeded',
+  'normal',
+  'in-progress',
+  'incomplete',
+  'not-started',
+  'pending',
+  'unknown',
+  'information',
+];
+
+const iconTypes = {
+  error: ErrorFilled,
+  ['caution-major']: WarningAltInvertedFilled,
+  ['caution-minor']: WarningAltFilled,
+  undefined: UndefinedFilled,
+  succeeded: CheckmarkFilled,
+  normal: CheckmarkOutline,
+  ['in-progress']: InProgress,
+  incomplete: Incomplete,
+  ['not-started']: CircleDash,
+  pending: PendingFilled,
+  unknown: UnknownFilled,
+  information: WarningSquareFilled,
+};
+
+export type IconIndicatorKind = (typeof IconIndicatorKinds)[number];
+
+interface IconIndicatorProps {
+  /**
+   * Specify an optional className to add.
+   */
+  className?: string;
+
+  /**
+   * Specify the kind of icon to be used
+   */
+  kind: IconIndicatorKind;
+
+  /**
+   * Label next to the icon
+   */
+  label: string;
+
+  /**
+   * Specify the size of the Icon Indicator. Currently supports either 16 (default) or 20 sizes.
+   */
+  size?: number;
```

Comment:
This can be more strict to ensure invalid sizes aren't given

```suggestion
  /**
   * Specify the size of the Icon Indicator. Defaults to 16.
   */
  size?: 16 | 20;
```